test/integration/node-specific/mongo_client.test.ts

```diff
@@ -0,0 +1,605 @@
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'client.close will not permit operations to auto reconnect',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        await client.db('test').collection('test').findOne();
+
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+        await client.close();
+
+        expect(client).to.have.property('topology', undefined);
+
+        const result = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(client).to.have.property('topology', undefined);
+        expect(result).to.be.instanceOf(MongoNotConnectedError);
+
+        await client.close();
+      }
+    );
+
+    it(
+      'client.close will not permit operations to auto reconnect permanently',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        expect(client.s).to.have.property('hasBeenClosed', false);
+
+        await client.db('test').collection('test').findOne();
+
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+        await client.close();
+
+        expect(client).to.have.property('topology', undefined);
+
+        const firstResult = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(client).to.have.property('topology', undefined);
+        expect(firstResult).to.be.instanceOf(MongoNotConnectedError);
+
+        const postCloseHasBeenClosedDescriptor = {
+          value: true,
+          enumerable: true,
+          configurable: false,
+          writable: false
+        };
+
+        expect(client.s).to.have.ownPropertyDescriptor(
+          'hasBeenClosed',
+          postCloseHasBeenClosedDescriptor
+        );
+
+        const pingCommandToBeStarted = once(client, 'commandStarted');
+        await client.connect(); // explicitly connect again
+        const [pingOnConnect] = await pingCommandToBeStarted;
+
+        expect(pingOnConnect).to.have.property('commandName', 'ping');
+
+        expect(client.s).to.have.ownPropertyDescriptor(
+          'hasBeenClosed',
+          postCloseHasBeenClosedDescriptor
+        );
+
+        await client.close();
+
+        const secondResult = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(client).to.have.property('topology', undefined);
+        expect(secondResult).to.be.instanceOf(MongoNotConnectedError);
+
+        expect(client.s).to.have.ownPropertyDescriptor(
+          'hasBeenClosed',
+          postCloseHasBeenClosedDescriptor
+        );
+      }
+    );
+  });
+});
```

Comment:
Just for visibility, let's make sure we don't close the actual jira ticket until we do this part of the AC:

> Review tests and update any that explicitly call MongoClient connect.
> File a separate PR for this.

Reply:
Yep, @baileympearson is on top of this, and I'll be posting the test PR after we merge this avoiding rebase annoyances.